source/Mole/GraphBuilder.class.st

```diff
@@ -1,32 +1,116 @@
 "
-I'm an abstract class aiding with the creation of graphs.
+I'm a builder of directed graphs
 "
 Class {
 	#name : #GraphBuilder,
 	#superclass : #Object,
+	#instVars : [
+		'vertices',
+		'labelByEdgeArray'
+	],
 	#category : #Mole
 }
 
-{ #category : #adding }
+{ #category : #configuring }
 GraphBuilder >> addVertex: aVertex [
 
-	self subclassResponsibility
+	vertices add: aVertex
 ]
 
-{ #category : #adding }
+{ #category : #configuring }
 GraphBuilder >> addVertices: aVertexCollection [
 
 	aVertexCollection do: [ :vertex | self addVertex: vertex ]
 ]
 
 { #category : #building }
-GraphBuilder >> build [
+GraphBuilder >> buildDirected [
 
-	^ self subclassResponsibility
+	| directedEdges |
+
+	directedEdges := Set new.
+	labelByEdgeArray
+		keysAndValuesDo:
+			[ :edgeArray :labelOptional | directedEdges add: (self directedEdgeFrom: edgeArray labeledAccordingTo: labelOptional) ].
+
+	^ DirectedGraph withValidVertices: vertices andEdges: directedEdges
+]
+
+{ #category : #building }
+GraphBuilder >> buildUndirected [
+
```

Comment:
buildDirected and buildUndirected are similar enough.. surely you can extract the common behavior if abstract the edge creation.